rs/interfaces/src/execution_environment.rs

```diff
@@ -1132,6 +1134,26 @@ pub trait SystemApi {
     /// Returns the amount of cycles added to the canister's balance.
     fn ic0_mint_cycles(&mut self, amount: u64) -> HypervisorResult<u64>;
 
+    /// Mints the `amount` cycles
+    /// Adds cycles to the canister's balance.
+    ///
+    /// Adds no more cycles than `amount`.
+    ///
+    /// The canister balance afterwards does not exceed
+    /// maximum amount of cycles it can hold.
+    /// However, canisters on system subnets have no balance limit.
```

Comment:
> canisters on system subnets have no balance limit

What does this sentence mean? Even on system subnets, I suppose cycles operations are saturating.

Reply:
> which is about 10^26 dollars worth of cycles

this argument might not apply for UTOPIA